### DISCUSSION TITLE
Custom Dashboards

### DISCUSSION THREAD

**Discussion by jannikjw-sai** [langchain-ai/langsmith-sdk#425] [Ideas]
### Feature request

Enable developers to build their own visualizations on LangSmith using matplotlib or plotly.

### Motivation

I would like to visualize certain performance metrics that are not sufficiently shown in the graphs in the LangSmith metrics. This way leaders and developers can understand what exactly is happening.

**Comment by hinthornw** [comment]
Hi @jannikjw-sai  - thanks for raising this issue! Would love to hear more about what kind of visualizations you'd find most useful. Especially if you are particularly interested in monitoring vs. testing/evaluation

**Comment by jannikjw-sai** [comment]
Hi @hinthornw, I am particularly interested in comparing different models and calculating different self-defined metrics in bar charts. I also want a time series of all my dataset runs. Mostly, I want to run quality evaluations.

**Comment by jannikjw-sai** [comment]
@hinthornw What is the timeline for this? I would love to have a coding interface to simply create my own dashboard.

**Comment by hinthornw** [comment]
Got it, @jannikjw-sai - thanks! We hope to get the bar charts for test results out in the next couple of weeks, and time series would be later.

We also have a convenience method in the client:
```python
client.get_test_results(project_name="my-test")
```

that you can use to fetch the results and metrics as a pandas dataframe. It's fairly straightforward to plot the results then from there
